presto-main/src/test/java/com/facebook/presto/operator/scalar/AbstractTestFunctions.java

```diff
@@ -198,6 +199,23 @@ protected void assertInvalidCast(String projection, String message)
         }
     }
 
+    protected void assertNotSupported(String projection, Type expectedType, String message)
```

Comment:
This is will call tryEvaluateWithAll, but one of the engine will actually return a valid type for the given expression and check if it matches with the expectedType, in FunctionAssertions.executeProjectionWithAll(). So there is not real way to do this without a type.

Also, there are existing methods like this in this class: assertNumericOverflow(), assertInvalidCast() ...